keyhive_core/src/keyhive.rs

```diff
@@ -1760,4 +1760,50 @@ mod tests {
 
         assert_eq!(dlg.delegation.subject_id(), doc.borrow().doc_id().into());
     }
+
+    #[tokio::test]
+    async fn cgka_ops_for_doc() {
+        // Create two keyhives, run membership sync (i.e. membersip operatiosn and prekey operations)
+        // then attempt to lookup cgka operations on the other_keyhive for the doc we just synced
+
+        // First create the two keyhives
+        let mut alice = make_keyhive().await;
+        let mut bob = make_keyhive().await;
+
+        // Now make a document on alice which bob can read
+        let bob_indi = Rc::new(RefCell::new(bob.active().borrow().individual().clone()));
+        let doc = alice
+            .generate_doc(vec![bob_indi.dupe().into()], nonempty![[0u8; 32]])
+            .await
+            .unwrap();
+
+        // Now pull out all the membership operations and prekey operations from alice
+        let mut events = alice.static_events_for_agent(&bob_indi.into()).unwrap();
+
+        // Now ingest all these events on bob
+        let mut iterations = 0;
+        loop {
+            iterations += 1;
+            if iterations > 100 {
+                panic!("Too many iterations");
+            }
+            let evts = std::mem::take(&mut events);
+            if evts.is_empty() {
+                break;
+            }
+            for (digest, op) in evts {
+                if let Err(_) = bob.receive_static_event(op.clone()) {
+                    events.insert(digest, op);
+                }
+            }
+        }
+
+        // Check the doc exists
+        bob.get_document(doc.borrow().doc_id()).unwrap();
+
+        // FIXME: Alex plz confirm no longer the case. Original comment as follows:
+        // This will throw an error because the heads of the cgka operation graph are empty
```

Comment:
I added an `assert!` below just to make the fix easier to confirm. If this is the behaviour that you expect, delete these lines!